frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/code_block/code_block_language_selector.dart

```diff
@@ -160,34 +163,129 @@ class _LanguageSelectionPopoverState extends State<_LanguageSelectionPopover> {
 
   @override
   Widget build(BuildContext context) {
-    return Column(
-      mainAxisSize: MainAxisSize.min,
-      children: [
-        FlowyTextField(
-          focusNode: focusNode,
-          autoFocus: false,
-          controller: searchController,
-          hintText: LocaleKeys.document_codeBlock_searchLanguageHint.tr(),
-          onChanged: (_) => setState(() {
-            filteredLanguages = widget.supportedLanguages
-                .where((e) => e.contains(searchController.text.toLowerCase()))
-                .map((e) => e.capitalize())
-                .toList();
-            selectedIndex =
-                widget.supportedLanguages.indexOf(widget.language ?? '');
-          }),
-        ),
-        const VSpace(8),
-        Flexible(
-          child: SelectableItemListMenu(
-            shrinkWrap: true,
-            items: filteredLanguages,
-            selectedIndex: selectedIndex,
-            onSelected: (index) =>
-                widget.onLanguageSelected(filteredLanguages[index]),
-          ),
+    return Shortcuts(
+      shortcuts: const {
+        SingleActivator(LogicalKeyboardKey.arrowUp): _ArrowUpIntent(),
+        SingleActivator(LogicalKeyboardKey.arrowDown): _ArrowDownIntent(),
+        SingleActivator(LogicalKeyboardKey.enter): _EnterIntent(),
+      },
+      child: Actions(
+        actions: {
+          _ArrowUpIntent: _ArrowUpAction(this),
+          _ArrowDownIntent: _ArrowDownAction(this),
+          _EnterIntent: _EnterAction(this),
```

Comment:
I prefer to use `CallbackAction` here to:

1. avoid passing the `_LanguageSelectionPopoverState`, and
2. avoid wrapping the `setState` in the `refresh` function

```
        actions: <Type, Action<Intent>>{
          _ArrowUpIntent: CallbackAction<_ArrowUpIntent>(
            onInvoke: _handleKeyInvoke,
          ),
          _ArrowDownIntent: CallbackAction<_ArrowDownIntent>(
            onInvoke: _handleKeyInvoke,
          ),
          _EnterIntent: CallbackAction<_EnterIntent>(
            onInvoke: _handleKeyInvoke,
          ),
        },
```